Sources/Components/Radio/Radio.swift

```diff
@@ -0,0 +1,136 @@
+        }
+    }
+    
+    /// A single radio button view used within `RadioButtonGroup`.
+    ///
+    /// Displays a selectable circular button with a label. When the button is selected,
+    /// the circle is filled to indicate the selection.
+    ///
+    /// - Parameters:
+    ///   - isSelected: A Boolean value indicating whether the radio button is selected.
+    ///   - label: The text label for the radio button.
+    ///   - state: The state of the radio button (default, error, disabled).
+    ///   - action: A closure that is executed when the radio button is tapped.
+    struct RadioButton: View {
+        /// A Boolean value indicating whether the radio button is selected.
+        var isSelected: Bool
+        /// The text label for the radio button.
+        var label: String
+        /// The state of the radio button (default, error, disabled).
+        var state: RadioButtonState
+        /// A closure that is executed when the radio button is tapped.
+        var action: () -> Void
+        /// Object that will provide needed colors.
+        private let colorProvider: ColorProvider = Warp.Color
+        
+        var body: some View {
+            HStack(spacing: Spacing.spacing100) {
+                Circle()
+                    .strokeBorder(borderColor, lineWidth: isSelected ? 6 : 1)
+                    .background(Circle().fill(fillColor))
+                    .frame(width: 20, height: 20)
+                SwiftUI.Text(label)
+                    .font(Typography.body.font)
+                    .foregroundColor(textColor)
+                Spacer()
+            }
+            .onTapGesture {
+                if state != .disabled {
+                    action()
+                }
+            }
+        }
+        
+        private var borderColor: Color {
+            switch (state, isSelected) {
+            case (.default, true):
+                return colorProvider.radioBorderSelected
+            case (.default, false):
+                return colorProvider.radioBorder
+            case (.error, true):
+                return colorProvider.radioNegativeBorder
+            case (.error, false):
+                return colorProvider.radioNegativeBorder
+            case (.disabled, true):
+                return colorProvider.radioBorderDisabled
+            case (.disabled, false):
+                return colorProvider.radioBorderDisabled
+            }
+        }
+        
+        private var fillColor: Color {
+            if state == .disabled {
+                colorProvider.radioBackgroundDisabled
+            } else {
+                .clear
+            }
+        }
+        
+        private var textColor: Color {
+            switch state {
+            case .default:
+                return colorProvider.token.text
+            case .error:
+                return colorProvider.token.text
+            case .disabled:
+                return colorProvider.token.textDisabled
+            }
+        }
+    }
```

Comment:
```suggestion
        private var textColor: Color {
            switch state {
            case .default, .error:
                return colorProvider.token.text
            case .disabled:
                return colorProvider.token.textDisabled
            }
        }
    }
```

I can't determine if I like the explicit-y of showing all states or combining the cases with the same output into one.

You decide 😄

Reply:
yeah I like the simplicity of your suggestion. When you are writing it you change things one by one so you don't see it later that they are the same color